src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,201 @@
+                result.add(convertToCar(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t read all data from DB", e);
+        }
+        result.forEach(c -> c.setDrivers(getDrivers(c.getId())));
+        return result;
+    }
+
+    @Override
+    public Car update(Car car) {
+        String query = "UPDATE cars SET model = ?, manufacturer_id = ? "
+                + "WHERE id = ? AND is_deleted = FALSE;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setString(1, car.getModel());
+            statement.setLong(2, car.getManufacturer().getId());
+            statement.setLong(3, car.getId());
+            statement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t update car with ID: " + car.getId(), e);
+        }
+        deleteDrivers(car.getId());
+        addDrivers(car);
+        return car;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "UPDATE cars SET is_deleted = TRUE WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            return statement.executeUpdate() >= 1;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t delete car with ID: " + id, e);
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long id) {
+        List<Car> result = new ArrayList<>();
+        String query = "SELECT c.id AS car_id, c.model, "
+                + "m.id AS manufacturer_id, m.name, m.country FROM cars c "
+                + "JOIN cars_drivers cd ON c.id = cd.car_id "
+                + "JOIN manufacturers m ON c.manufacturer_id = m.id  "
+                + "WHERE cd.driver_id = ? AND c.is_deleted = FALSE AND m.is_deleted = FALSE;;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                result.add(convertToCar(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t get all cars by driver with ID: " + id, e);
+        }
+        result.forEach(c -> c.setDrivers(getDrivers(c.getId())));
+        return result;
+    }
+
+    private List<Driver> getDrivers(Long carId) {
+        List<Driver> result = new ArrayList<>();
+        String query = "SELECT id, name, license_number FROM drivers d "
+                + "JOIN cars_drivers cd ON d.id = cd.driver_id "
+                + "WHERE cd.car_id = ? AND c.id_deleted = FALSE"
```

Comment:
Does it work without space?
```suggestion
                + "WHERE cd.car_id = ? AND c.id_deleted = FALSE "
```